app/src/lib/stores/pull-request-store.ts

```diff
@@ -13,136 +13,181 @@ import {
+    account: Account
+  ): Promise<void> {
+    const api = API.fromAccount(account)
+
+    const statuses: Array<IPullRequestStatus> = []
+    const prs: Array<PullRequest> = []
 
-    const builtPullRequests = new Array<PullRequest>()
     for (const pr of pullRequests) {
-      const headId = pr.head.repoId
-      let head: GitHubRepository | null = null
-      if (headId) {
-        head = await this.repositoriesStore.findGitHubRepositoryByID(headId)
+      const apiStatus = await api.fetchCombinedRefStatus(
+        repository.owner.login,
+        repository.name,
+        pr.head.sha
+      )
+
+      const status = {
+        pullRequestNumber: pr.number,
+        state: apiStatus.state,
+        totalCount: apiStatus.total_count,
+        sha: pr.head.sha,
       }
 
-      // We know the base repo ID can't be null since it's the repository we
-      // fetched the PR from in the first place.
-      const baseId = forceUnwrap(
-        'PR cannot have a null base repo id',
-        pr.base.repoId
-      )
-      const base = forceUnwrap(
-        'PR cannot have a null base repo',
-        await this.repositoriesStore.findGitHubRepositoryByID(baseId)
-      )
+      statuses.push({
+        pullRequestId: pr.id,
+        state: apiStatus.state,
+        totalCount: apiStatus.total_count,
+        sha: pr.head.sha,
+      })
 
-      // We can be certain the PR ID is valid since we just got it from the
-      // database.
-      const prID = forceUnwrap(
-        'PR cannot have a null ID after being retrieved from the database',
-        pr.id
+      prs.push(
+        new PullRequest(
+          pr.id,
+          pr.created,
+          status,
+          pr.title,
+          pr.number,
+          pr.head,
+          pr.base,
+          pr.author
+        )
       )
-      const prStatus = await this.getPullRequestStatusById(pr.head.sha, prID)
+    }
 
-      const builtPR = new PullRequest(
-        prID,
-        new Date(pr.createdAt),
-        prStatus,
-        pr.title,
-        pr.number,
-        new PullRequestRef(pr.head.ref, pr.head.sha, head),
-        new PullRequestRef(pr.base.ref, pr.base.sha, base),
-        pr.author
-      )
+    await this.writePRStatus(statuses)
+
+    this.pullRequestStatuses = await this.getPRStatuses(
+      repository,
+      pullRequests
+    )
+    this.emitUpdate()
+  }
+
+  private async getPRStatusById(
+    sha: string,
+    pullRequestId: number
+  ): Promise<PullRequestStatus | null> {
+    const result = await this.pullRequestDatabase.pullRequestStatus
+      .where('[sha+pullRequestId]')
+      .equals([sha, pullRequestId])
+      .limit(1)
+      .first()
 
-      builtPullRequests.push(builtPR)
+    if (!result) {
+      return null
     }
 
-    return builtPullRequests
+    return new PullRequestStatus(
+      result.pullRequestId,
+      result.state,
+      result.totalCount,
+      result.sha
+    )
   }
 
-  private async writePullRequests(
+  private async writePRs(
```

Comment:
:lipstick: is there a reason why we've abbreviated some methods to `PRs` in some places but not others?

Reply:
Yes, the public API uses the full name `PullRequest` and internal methods are abbreviated.